internal/uvm/update_uvm.go

```diff
@@ -42,6 +44,10 @@ func (uvm *UtilityVM) UpdateConstraints(ctx context.Context, data interface{}, a
 				processorLimits.Weight = uint64(*resources.CPU.Shares)
 			}
 		}
+	case *ctrdtaskapi.PolicyFragment:
+		return uvm.InjectPolicyFragment(ctx, resources)
+	default:
+		return fmt.Errorf("invalid resource: %v", resources)
 	}
 
 	if memoryLimitInBytes != nil {
```

Comment:
It won't let me comment on line 65 below, but I'm curious why CPU group was added via annotations instead of the new type approach we took here.

Reply:
hm. not sure. maybe at that time we didn't consider extending task resource types due to potential difficulties upstreaming. I guess it can be changed now. But maybe @katiewasnothere can answer better here 😄

Reply:
Curious to hear what @katiewasnothere's opinion is here before we merge.

Reply:
Iirc, I think Maksim is correct here. I was trying to keep consistent with upstream. In retrospect, it would have been fine to do this same thing here for cpu groups. There was no specific reason I didn't do that.

Reply:
Thanks, this approach seems fine then.